### ISSUE TITLE
Bug: Smart Lock not allowing lock/unlock though status works correctly

### ISSUE THREAD

**Issue by xhi-nico** [OpenWonderLabs/homebridge-switchbot#462]
### Describe The Bug

Smart lock lock/unlock functionality in HomeKit not working

### To Reproduce

add smart lock and attempt to lock/unlock

### Expected behavior

lock/unlock successfully

### Relevant log output

```shell
[06/09/2022, 16:16:42] [SwitchBot] [DEBUG] Lock: Apartment Lock Set LockTargetState: 0
[06/09/2022, 16:16:42] [SwitchBot] Lock: Apartment Lock Sending request to SwitchBot API. command: unlock, parameter: default, commandType: command
[06/09/2022, 16:16:43] [SwitchBot] [DEBUG] Lock: Apartment Lock pushchanges: {"statusCode":190,"body":{},"message":"not support device type"}
[06/09/2022, 16:16:43] [SwitchBot] Lock: Apartment Lock Device internal error due to device states not synchronized with server, Or command: {"statusCode":190,"body":{},"message":"not support device type"} format is invalid
[06/09/2022, 16:16:48] [SwitchBot] [DEBUG] Lock: Apartment Lock refreshStatus: {"statusCode":100,"body":{"deviceId":"D64AB0850524","deviceType":"Smart Lock","hubDeviceId":"C7E3699B113A","lockState":"locked","doorState":"closed","calibrate":true},"message":"success"}
[06/09/2022, 16:16:48] [SwitchBot] [DEBUG] Lock: Apartment Lock On: 1
[06/09/2022, 16:16:48] [SwitchBot] [DEBUG] Lock: Apartment Lock updateCharacteristic LockTargetState: 1
[06/09/2022, 16:16:48] [SwitchBot] [DEBUG] Lock: Apartment Lock updateCharacteristic LockCurrentState: 1
[06/09/2022, 16:17:17] [SwitchBot] [DEBUG] Lock: Apartment Lock Set LockTargetState: 0
[06/09/2022, 16:17:17] [SwitchBot] Lock: Apartment Lock Sending request to SwitchBot API. command: unlock, parameter: default, commandType: command
[06/09/2022, 16:17:17] [SwitchBot] [DEBUG] Lock: Apartment Lock pushchanges: {"statusCode":190,"body":{},"message":"not support device type"}
[06/09/2022, 16:17:17] [SwitchBot] Lock: Apartment Lock Device internal error due to device states not synchronized with server, Or command: {"statusCode":190,"body":{},"message":"not support device type"} format is invalid
[06/09/2022, 16:17:23] [SwitchBot] [DEBUG] Lock: Apartment Lock refreshStatus: {"statusCode":100,"body":{"deviceId":"D64AB0850524","deviceType":"Smart Lock","hubDeviceId":"C7E3699B113A","lockState":"unlocked","doorState":"closed","calibrate":true},"message":"success"}
[06/09/2022, 16:17:23] [SwitchBot] [DEBUG] Lock: Apartment Lock On: 0
[06/09/2022, 16:17:23] [SwitchBot] [DEBUG] Lock: Apartment Lock updateCharacteristic LockTargetState: 0
[06/09/2022, 16:17:23] [SwitchBot] [DEBUG] Lock: Apartment Lock updateCharacteristic LockCurrentState: 0
```

### Config for homebridge-switchbot

```shell
"options": {
                "devices": [
                    {
                        "deviceId": "--",
                        "configDeviceName": "Apartment Lock",
                        "offline": true,
                        "configDeviceType": "Smart Lock",
                        "lock": {
                            "hide_contactsensor": true
                        },
                        "firmware": "5.6",
                        "logging": "debug"
                    },
                    {
                        "deviceId": "--",
                        "configDeviceName": "Door Buzzer",
                        "offline": true,
                        "configDeviceType": "Bot",
                        "bot": {
                            "mode": "press",
                            "deviceType": "door",
                            "allowPush": true
                        },
                        "logging": "standard"
                    }
                ]
            },
            "platform": "SwitchBot"
        }
```

### Device & Model

SwitchBot Smart Lock

### Node.js Version

latest

### NPM Version

latest

### Homebridge Version

latest

### Homebridge Switchbot Plugin Version

latest

### Homebridge Config UI X Plugin Version

latest

### Operating System

Raspbian

**Comment by jamesalmeida** [comment]
Having same issue

**Comment by jwolkenhauer** [comment]
@GeorgeeNico @jamesalmeida   I was having this issue too.  Found the current workaround of using the SmartThings app and plugin works.  Check out this thread for more info.  (https://github.com/OpenWonderLabs/SwitchBotAPI/issues/121

**Comment by Jacksonbm1** [comment]
This is basically a duplicate of this issue I filed: https://github.com/OpenWonderLabs/homebridge-switchbot/issues/435

The issue has finally been fixed at the api level and hopefully gets fixed here: https://github.com/OpenWonderLabs/homebridge-switchbot/issues/435

**Comment by donavanbecker** [comment]
See https://github.com/OpenWonderLabs/homebridge-switchbot/issues/476

**Comment by donavanbecker** [comment]
Fixed with v2.0.0, Once released.
